yt_dlp/extractor/discoveryplusindia.py

```diff
@@ -0,0 +1,99 @@
+        headers['x-disco-client'] = 'WEB:UNKNOWN:dplus-india:17.0.0'
+
+    def _download_video_playback_info(self, disco_base, video_id, headers):
+        return self._download_json(
+            disco_base + 'playback/v3/videoPlaybackInfo',
+            video_id, headers=headers, data=json.dumps({
+                'deviceInfo': {
+                    'adBlocker': False,
+                },
+                'videoId': video_id,
+            }).encode('utf-8'))['data']['attributes']['streaming']
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        return self._get_disco_api_info(
+            url, display_id, 'ap2-prod-direct.discoveryplus.in', 'dplusindia', 'in')
+
+
+class DiscoveryPlusIndiaShowIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?discoveryplus\.in/show/(?P<show_name>\S*)'
+    _TESTS = [{
+        'url': 'https://www.discoveryplus.in/show/how-do-they-do-it',
+        'playlist_mincount': 140,
+        'info_dict': {
+            'id': 'how-do-they-do-it',
+        },
+    }
+    ]
+
+    def _entries(self, show_name):
+        headers = {
+            'authority': 'ap2-prod-direct.discoveryplus.in',
+            'x-disco-client': 'WEB:UNKNOWN:dplus-india:prod',
+            'x-disco-params': 'realm=dplusindia',
+            'origin': 'https://www.discoveryplus.in',
+            'referer': 'https://www.discoveryplus.in/',
+        }
+        show_url = 'https://ap2-prod-direct.discoveryplus.in/cms/routes/show/{}?include=default'.format(show_name)
+        show_json = self._download_json(show_url,
+                                        video_id=show_name,
+                                        headers=headers).get('included')[4].get('attributes').get('component')
+        show_id = show_json.get('mandatoryParams').split('=')[-1]
+        season_url = 'https://ap2-prod-direct.discoveryplus.in/content/videos?sort=episodeNumber&filter[seasonNumber]={}&filter[show.id]={}&page[size]=100&page[number]={}'
+        for season in show_json.get('filters')[0].get('options'):
+            season_id = season.get('id')
+            season_json = self._download_json(season_url.format(season_id, show_id, '1'), video_id=show_id, headers=headers)
+            total_pages = int(season_json.get('meta').get('totalPages')) + 1
+            for page in range(1, total_pages):
+                episode_url = season_url.format(season_id, show_id, str(page))
+                episodes_json = self._download_json(
+                    episode_url, video_id=show_id, headers=headers,
+                    note='Downloading JSON metadata page %s' % page).get('data')
+                for episode in episodes_json:
+
+                    video_id = episode.get('attributes').get('path')
+                    yield self.url_result(
+                        'https://discoveryplus.in/videos/%s' % video_id,
+                        ie=DiscoveryPlusIndiaIE.ie_key(), video_id=video_id)
```

Comment:
```suggestion            page_num = total_pages = 0
            while page_num <= total_pages:
                if page_num == 0:
                    total_pages = try_get(season_json, lambda x: x['meta']['totalPages'], int) or 0
                episodes_json = self._download_json(
                    season_url.format(season_id, show_id, compat_str(page_num + 1)), show_id,
                    note='Downloading JSON metadata%s' % (' page %d' % page_num if page_num else ''),
                    headers=headers)['data']
                for episode in episodes_json:
                    video_id = episode['attributes']['path']
                    yield self.url_result(
                        'https://discoveryplus.in/videos/%s' % video_id,
                        ie=DiscoveryPlusIndiaIE.ie_key(), video_id=video_id)
                page_num += 1
```

1. No need to download the same json twice
2. Page numbers start from 0. The 0th page just doesn't show the `page %d` part. That is how the other extractors count
3. Use try_get and compat_str

Reply:
> Page numbers start from 0.
Are you talking about yt-dlp, because with the api, requesting for 0th page throws an error 
`{
    "errors": [
        {
            "status": "400",
            "code": "invalid.payload",
            "id": "Cv8uaoYS9W",
            "detail": "Page number can not be smaller than 1"
        }
    ]
}`

Reply:
Yes, I am talking about the page numbers shown to user. When calling api, notice that I am using `compat_str(page_num + 1)`